parquet-hadoop/src/main/java/org/apache/parquet/ParquetReadOptions.java

```diff
@@ -333,13 +406,17 @@ public Builder copy(ParquetReadOptions options) {
 
     public ParquetReadOptions build() {
       if (codecFactory == null) {
-        codecFactory = HadoopCodecs.newFactory(0);
+        if (conf == null) {
+          codecFactory = HadoopCodecs.newFactory(0);
+        } else {
+          codecFactory = HadoopCodecs.newFactory(conf, 0);
```

Comment:
It seems that original ParquetReadOptions does not require any Configuration parameter. Should we avoid adding this by using ParquetConfiguration internally in the HadoopCodecs?

Reply:
We still need a way to get the ParquetConfiguration to the HadoopCodecs, for this I used a field in ParquetReadOptions. To this effect, I would also like to start a conversation in the future about the options classes, as it seems to me we can replace them with the more versatile ParquetConfiguration in most, if not all places. This as I do think this current flow of passing the ParquetConfiguration through the ParquetReadOptions is a bit hacky.